tests/Commands/DrupalCommandsTest.php

```diff
@@ -2,71 +2,82 @@
-    // Setup test directory.
-    $sites_subdir = isset($config['drupal']['site']['sites_subdir']) ? $config['drupal']['site']['sites_subdir'] : 'default';
-    $settings_root = $this->getSandboxRoot() . '/build/sites/default';
-    mkdir($settings_root, 0777, TRUE);
+    /**
+     * Test Toolkit very own "drupal:settings-setup" command.
+     *
+     * @param array $config
+     *   A configuration array.
+     * @param mixed $initial_default_settings
+     *   A initial default settings.
+     * @param mixed $initial_settings
+     *   A initial settings.
+     * @param array $expected
+     *   Test assertions.
+     *
+     * @dataProvider dataProvider
+     */
+    public function testDrupalSettingsSetup(array $config, $initial_default_settings, $initial_settings, array $expected) {
+      // Setup test Task Runner configuration file.
+      $configFile = $this->getSandboxFilepath('runner.yml');
+      file_put_contents($configFile, Yaml::dump($config));
 
-    // Setup initial default.settings.php and settings.php, if any.
-    file_put_contents($settings_root . '/default.settings.php', $initial_default_settings);
+      // Setup test directory.
+      $sites_subdir = isset($config['drupal']['site']['sites_subdir']) ? $config['drupal']['site']['sites_subdir'] : 'default';
+      $settings_root = $this->getSandboxRoot() . '/build/sites/default';
+      mkdir($settings_root, 0777, TRUE);
 
-    // Setup settings.php file, if test case requires it.
-    if ($initial_settings) {
-      file_put_contents($settings_root . '/settings.php', $initial_settings);
-    }
+      // Setup initial default.settings.php and settings.php, if any.
+      file_put_contents($settings_root . '/default.settings.php', $initial_default_settings);
+
+      // Setup settings.php file, if test case requires it.
+      if ($initial_settings) {
+        file_put_contents($settings_root . '/settings.php', $initial_settings);
+      }
 
-    // Run command.
-    $input = new StringInput('drupal:settings-setup --working-dir=' . $this->getSandboxRoot());
-    $runner = new TaskRunner($input, new BufferedOutput(), $this->getClassLoader());
-    $runner->run();
+      // Run command.
+      $input = new StringInput('drupal:settings-setup --working-dir=' . $this->getSandboxRoot());
+      $runner = new TaskRunner($input, new BufferedOutput(), $this->getClassLoader());
+      $runner->run();
 
-    // Assert expectations.
-    foreach ($expected as $row) {
-      $content = file_get_contents($this->getSandboxFilepath($row['file']));
-      $this->assertContainsNotContains($content, $row);
+      // Assert expectations.
+      foreach ($expected as $row) {
+        $content = file_get_contents($this->getSandboxFilepath($row['file']));
+        $this->assertContainsNotContains($content, $row);
+      }
     }
+
+  }
+}
+
+namespace EcEuropa\Toolkit\TaskRunner\Commands {
+
+  /**
+   * Override random_bytes function for test.
+   */
+  function random_bytes() {
+    return 'abc';
```

Comment:
Shouldn't this be `return 'YWJj'`?

Reply:
Because it's the result of `str_replace(['+', '/', '='], ['-', '_', ''], base64_encode('abc'))`.

Reply:
right! :D